### ISSUE TITLE
Add guidelines for use of skos in OMO

### ISSUE THREAD

**Issue by cmungall** [information-artifact-ontology/ontology-metadata#65]
I have heard various people say that skos should not be used in conjunction with OWL. One reason is that formally domain and range constraints are skos:Concept, which introduces punning.

I think the solution is that we simply include the subset of skos we need in OMO, declared as APs, and recommend that this is not merged with skos.owl. However, we need to be able to anticipate objections. I think we need a page about this in the OMO reference guide.

**Comment by matentzn** [comment]
Where was this conversation had? Any tickets you could link? 

I at least would like to challenge this, but happy to roll once I understand why I should not be using skos:relatedMatch. (I know only the usual objections of skos:exactMatch vs owl:equivalentClass).

The punning issue is a bit annoying but I am kind of half way happy to ignore this.. What are the implications of this pun?

**Comment by matentzn** [comment]
I am pretty sure neither the OWLAPI nor OWL reasoners would care about SKOS at all.. So practically speaking, what tool/framework is known to actually do this punning?

**Comment by graybeal** [comment]
They care about any odd/inconsistent modeling. They know nothing about SKOS. If what you're saying is no one would apply them to primarily SKOS artifacts, I agree it seems unlikely.

**Comment by cmungall** [comment]
I just realized there is this old W3C note, but it predates OWL2:

https://www.w3.org/2006/07/SWD/SKOS/skos-and-owl/master.html

> So practically speaking, what tool/framework is known to actually do this punning?

not sure I understand. Take

http://www.w3.org/TR/skos-reference/skos.rdf

pass it through the OWLAPI, you get ObjectProperties for exactMatch and friends. Combine this with an ontology that uses exactmatch on a class and you get punning. Is this a technical problem? No. Is it confusing? Probably.

I think the recommendation here is that you simply don't mix skos.rdf and OBO. If you do then it's on you if you get punning. If you want axioms and declarations on skos properties (which is a reasonable thing to want) then I think it behooves us to provide this, e.g. something like /obo/omo/skos.owl.

Will that confuse people? Well if you mix the two skos renderings then you get punning on APs and OPs.

**Comment by matentzn** [comment]
I did not know that! I thought they were just treated like APs.

**Comment by matthewhorridge** [comment]
If the properties are declared as annotation properties, locally, then this would avoid the punning.  Perhaps confusing... not sure

**Comment by cmungall** [comment]
yep, would avoid the (legal but confusing) punning on the classes that are
subject/object of the triples...

but potential (illegal) punning if omo.owl and skos.rdf are ever merged or
both imported to the same ontology? Would anyone ever want to do that? We
can simply say: don't do that. You are in OBO, these are things that are
permitted to be combined. I don't know if it would create obstacles to
uptake later

maybe the 3rd way: no commitment from OMO. No punning by default, but if
someone does combine skos and omo and an OBO the worse they get is (legal)
class punning?

**Comment by cmungall** [comment]
Thanks!

1: if it has no formal semantics, then we model as APs and there is no problem of conflicting semantics
2: I think the undecidability is a red herring (but I differ with many DL people here). I am trying to think of a real-world scenario where we run into problems. And if we model as APs then there is no punning in the first place

IMHO the main challenge is more of a social-governance one. Everything just works if we model skos as APs. But this is not officially sanctioned by SKOS (I don't know if it's explicitly discouraged). And we can't prohibit people from combining the official skos.rdf with OBO which would result in punning - this is not worrying from the decidability aspect, more just from confusion in how tooling presents punning. E.g. how would such a merged ontology look in OLS?

Perhaps we could add one step to my 4 steps above wjich is to engage the broader community about an officially sanctioned variant of SKOS that commits to APs for the IRIs (we would pursue in parallel we would not wait on this)
